### PR TITLE
add index on s_order_basket.userID

### PR DESCRIPTION
**Pull request by jonashrem**
### 1. Why is this change necessary?
we have noticed shopware doing querries like this one on many of our customers:
```
UPDATE `s_order_basket` SET `sessionID` = ? WHERE `userID` = ? AND `modus` = ? 
```

yet, this querry is not efficieent as there is no index on userID.

We noticed that this querry can create quite a load on the database on high traffic szenarios.

### 2. What does this change do, exactly?

Add an MySQL index to s_order_basket.userID

### 3. Describe each step to reproduce the issue or behaviour.
/

### 4. Please link to the relevant issues (if any).
/

### 5. Which documentation changes (if any) need to be made because of this PR?
/

### 6. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by JoshuaBehrens** [comment]
Aaah something I can drop from my bucket list 😁

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+class Migrations_Migration966 Extends Shopware\Components\Migrations\AbstractMigration
```
Comment (shyim) [inline _sql/migrations/966-add-userid-index-to-s-order-basket.php]:
```suggestion
class Migrations_Migration1710 extends Shopware\Components\Migrations\AbstractMigration
```

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+{
+    public function up($modus)
```
Comment (shyim) [inline _sql/migrations/966-add-userid-index-to-s-order-basket.php]:
```suggestion
    public function up($modus): void
```

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php (hunk 1 above)
Comment (shyim) [reply _sql/migrations/966-add-userid-index-to-s-order-basket.php]: Don't forget to change the file name too :)

**Comment by shyim** [comment]
Looks good. Can you do apply these two changes?

**Comment by jonashrem** [comment]
> Looks good. Can you do apply these two changes?

Thanks. I just applied them.

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php (hunk 1 above)
Comment (jonashrem) [reply _sql/migrations/966-add-userid-index-to-s-order-basket.php]: So filename still needs to be changed to 1710-add-userid-index-to-s-order-basket.php as far as I understand. Is this correct? @shyim

Comment (shyim) [reply _sql/migrations/966-add-userid-index-to-s-order-basket.php]: Yes

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+class Migrations_Migration1710 extends Shopware\Components\Migrations\AbstractMigration
```
Comment (mitelg) [inline _sql/migrations/966-add-userid-index-to-s-order-basket.php]: the filename still needs to be adjusted :blush:

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php (hunk 1 above)
Comment (jonashrem) [reply _sql/migrations/966-add-userid-index-to-s-order-basket.php]: I just changed the filename

File: _sql/migrations/966-add-userid-index-to-s-order-basket.php (hunk 3 above)
Comment (jonashrem) [reply _sql/migrations/966-add-userid-index-to-s-order-basket.php]: I just changed it

**Comment by shyim** [comment]
Thanks for your Contribution! :blue_heart:

**Pull request closed without merging**